### PR TITLE
Closes #574 Adding az_menu block_content type.

### LINKED ISSUE #574: Create Menu block block type
Match Quickstart 1 implementation. Should be added in Content > Blocks.
- Have menu selection
- Allow selection of start level
- Allow selection of expand children
- Should be embeddable in paragraphs
- Should be organizable in Structure > Blocks

QS1 version 
image
image

### PR DESCRIPTION
**Pull request by trackleft**
## Related issues
Closes #574
Please review https://git.drupalcode.org/project/field_menu/-/merge_requests/2/diffs

## How to test

Enable the `az_block_types_menu` module. (Not enabled by default)
image
Go to: /admin/content/blocks
image
See the menu option.
Go to `/admin/structure/block/block-content/manage/az_menu`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/fields`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/display`
Review settings 
Go to `/admin/structure/block/block-content/manage/az_menu/auto-label`
Review settings 
Either add a new menu at `/admin/structure/menu/add`
or 
Go to `/block/add/az_menu`
 
Select your new menu or any of the other existing menus in the Root field of the menu field set.
You should be able to select any level of any menu.
Save your new block

Go to `/node/add/az_flexible_page`
Add a title and a new "text" paragraph item.
Then click the block embed icon in the editor
Search for your block by the description you added when creating a new block.
When. found, click next and choose some options.
Your menu block should now be embedded within a text area.

Now that you know how to add a new menu block that can be embedded, be creative and add menu blocks to any field that allows embedding blocks.

Additionally you should be able to add the new block via the traditional block layout page.

Uninstall `az_block_types_menu` module to delete all of your tests.
Reinstall `az_block_types_menu` module to ensure it is possible.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
See https://558857f5-a288-42e0-95b6-71f0ce74e742.probo.build/menu-block-page

**Comment by danahertzberg** [comment]
How do I use/test this?

**Comment by danahertzberg** [comment]
This is looking great! Please see my test page: https://7af9e6f4-825e-471e-9baf-d4ecd0dbb84e.probo.build/dana

1. Background color is overriding text color of menu
2. Can the menu title style be updated to match the existing sidebar menu title style? (Wrapped in an `h4` class)

**Comment by trackleft** [comment]
#1 https://github.com/az-digital/az_quickstart/issues/1443

**Comment by akslay** [comment]
@trackleft Adding `.bg-white` to the `ul.nav` will address the text color issue.

```<ul class="clearfix nav flex-column nav-pills list-group bg-white">```

I am implementing the same fix here: https://github.com/az-digital/az_quickstart/pull/1624 .

File: themes/custom/az_barrio/templates/navigation/menu--block-content--az-menu--full--field-menu.html.twig [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+{#
+  We call a macro which calls itself to render the full tree.
+  @see http://twig.sensiolabs.org/doc/tags/macro.html
+#}
+{{ menus.menu_links(items, attributes, 0) }}
+{% macro menu_links(items, attributes, menu_level) %}
+  {% import _self as menus %}
+  {% if items %}
+    {% if menu_level == 0 %}
+      <ul{{ attributes.addClass('nav').addClass('flex-column').addClass('nav-pills').addClass('list-group') }}>
```
[the remark this answers is not in the document]
Comment (akslay) [reply themes/custom/az_barrio/templates/navigation/menu--block-content--az-menu--full--field-menu.html.twig:32]: @trackleft Adding .bg-white to the ul.nav will address the text color issue.

I am implementing the same fix here: https://github.com/az-digital/az_quickstart/pull/1624 .

**Comment by trackleft** [comment]
Should now appear like this (in the editor) because of the path following functionality.
image

**Comment by trackleft** [comment]
Should now appear like this on the menu block demo content page 
image

**Comment by trackleft** [comment]
The new block edit form looks like this 
image

**Comment by trackleft** [comment]
Drupal.org merge request with update function for upgrading from 2.0.x for field_menu module https://git.drupalcode.org/project/field_menu/-/merge_requests/2/diffs

**Comment by trackleft** [comment]
Closing in favor of #1887

**Pull request closed without merging**